shared-bindings/busio/SPI.c

```diff
@@ -276,7 +276,7 @@ STATIC mp_obj_t busio_spi_readinto(size_t n_args, const mp_obj_t *pos_args, mp_m
     normalize_buffer_bounds(&start, args[ARG_end].u_int, &length);
 
     if (length == 0) {
-        mp_raise_ValueError("Buffer must be at least length 1");
+        return mp_const_none;
```

Comment:
Correct the doc at line 251. Maybe say: "If the buffer or slice is of zero length, nothing will happen." or some better phrasing.